File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        visited_dict = dict()
        
        for index in range(len(nums)):
            num_to_target = target - nums[index]

            if num_to_target in visited_dict:
                return [visited_dict[num_to_target], index]

            visited_dict[nums[index]] = index

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        for (int num : nums){
            num = abs(num);
            if (nums[num] < 0)
                return num;
            nums[num] *= -1;
        }
        return -1;
    }
};
